aten/src/ATen/test/basic.cpp

```diff
@@ -361,24 +361,6 @@ TEST(BasicTest, FactoryMethodsTest) {
   ASSERT_FALSE(tensor0.requires_grad());
   ASSERT_FALSE(tensor0.is_pinned());
 
-  // Test setting requires_grad to false.
-  tensor0 = at::empty({4}, at::TensorOptions().requires_grad(false));
-  ASSERT_EQ(tensor0.dtype(), at::kFloat);
-  ASSERT_EQ(tensor0.layout(), at::kStrided);
-  ASSERT_EQ(tensor0.device(), at::kCPU);
-  ASSERT_FALSE(tensor0.requires_grad());
-  ASSERT_FALSE(tensor0.is_pinned());
-
-  // Test setting requires_grad to true.
-  tensor0 = at::empty({4}, at::TensorOptions().requires_grad(true));
-  ASSERT_EQ(tensor0.dtype(), at::kFloat);
-  ASSERT_EQ(tensor0.layout(), at::kStrided);
-  ASSERT_EQ(tensor0.device(), at::kCPU);
-  // This is a bug. Requires_grad was set to TRUE but this is being ignored.
-  // Issue https://github.com/pytorch/pytorch/issues/30405
-  ASSERT_FALSE(tensor0.requires_grad());
-  ASSERT_FALSE(tensor0.is_pinned());
```

Comment:
My recommendation is we fix #30405 first and then revisit this patch.

Reply:
I changed it to keep the requires_grad(false) test as is and changed the requires_grad(true) test to an EXPECT_ANY_THROW.